flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -289,20 +297,44 @@ def real_nodes(self):
     def real_edges(self):
         """Returns an iterable of all edges excluding the ones connected to the
         'low' or 'high' points."""
-        is_high_or_low = lambda n: n in ("low", "high")
         return filter(
-            lambda edge: not (is_high_or_low(edge[0]) or is_high_or_low(edge[1])),
+            lambda edge: edge[0] not in ("low", "high") and edge[1] not in ("low", "high"),
             self.edges(),
         )
 
+    def assign_weights(self, code):
+        """Assign the weights to the graph based on the weight
+        of the common vertex of each stabilizer pair of the code."""
+        for edge in self.edges():
+            data = self.edge_data(*edge)
+            if data["common_vertex"] is not None:
+                data["weight"] = code.graph.nodes[data["common_vertex"]].get("weight")
+            elif "high" in edge or "low" in edge:
+                data["weight"] = 0
```

Comment:
Should this be 0.0?

Reply:
Some algorithms required an int while some a float. Since we can always convert an int to a float but not the reverse, I think this is safer